x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx

```diff
@@ -16,138 +18,104 @@ import {
-
-  const { urlParams } = useUrlParams();
-
-  const { transactionName } = urlParams;
-
-  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
-
   const displayLog = uiSettings.get<boolean>(enableInspectEsQueries);
 
-  const {
-    ccsWarning,
-    log,
-    error,
-    histograms,
-    percentileThresholdValue,
-    isRunning,
-    progress,
-    startFetch,
-    cancelFetch,
-    overallHistogram,
-  } = useTransactionLatencyCorrelationsFetcher();
-
-  const startFetchHandler = useCallback(() => {
-    startFetch({
-      environment,
-      kuery,
-      serviceName,
-      transactionName,
-      transactionType,
-      start,
-      end,
+  const { progress, response, startFetch, cancelFetch } = useSearchStrategy(
+    APM_SEARCH_STRATEGIES.APM_LATENCY_CORRELATIONS,
+    {
       percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
-    });
-  }, [
-    startFetch,
-    environment,
-    serviceName,
-    transactionName,
-    transactionType,
-    kuery,
-    start,
-    end,
-  ]);
-
-  useEffect(() => {
-    startFetchHandler();
-    return cancelFetch;
-  }, [cancelFetch, startFetchHandler]);
+      analyzeCorrelations: true,
+    }
+  );
+  const progressNormalized = progress.loaded / progress.total;
+  const { overallHistogram, hasData, status } = getOverallHistogram(
+    response,
+    progress.isRunning
+  );
 
   useEffect(() => {
-    if (isErrorMessage(error)) {
+    if (isErrorMessage(progress.error)) {
       notifications.toasts.addDanger({
         title: i18n.translate(
           'xpack.apm.correlations.latencyCorrelations.errorTitle',
           {
             defaultMessage: 'An error occurred fetching correlations',
           }
         ),
-        text: error.toString(),
+        text: progress.error.toString(),
       });
     }
-  }, [error, notifications.toasts]);
+  }, [progress.error, notifications.toasts]);
 
   const [
     selectedSignificantTerm,
     setSelectedSignificantTerm,
-  ] = useState<MlCorrelationsTerms | null>(null);
+  ] = useState<LatencyCorrelation | null>(null);
 
   const selectedHistogram = useMemo(() => {
-    let selected = histograms.length > 0 ? histograms[0] : undefined;
-
-    if (histograms.length > 0 && selectedSignificantTerm !== null) {
-      selected = histograms.find(
+    let selected =
+      Array.isArray(response.latencyCorrelations) &&
+      response.latencyCorrelations.length > 0
+        ? response.latencyCorrelations[0]
+        : undefined;
+
+    if (
+      Array.isArray(response.latencyCorrelations) &&
+      selectedSignificantTerm !== null
+    ) {
+      selected = response.latencyCorrelations.find(
         (h) =>
-          h.field === selectedSignificantTerm.fieldName &&
-          h.value === selectedSignificantTerm.fieldValue
+          h.fieldName === selectedSignificantTerm.fieldName &&
+          h.fieldValue === selectedSignificantTerm.fieldValue
       );
     }
     return selected;
```

Comment:
You can avoid most of the branching (conditionals and ternaries) like this:

```ts
return response.latencyCorrelations?.find(
    (h) =>
      h.fieldName === selectedSignificantTerm?.fieldName &&
      h.fieldValue === selectedSignificantTerm?.fieldValue
  ) ?? response.latencyCorrelations?.[0];
```

Reply:
Updated in 6af23f3.